app/src/main/java/dev/andrylat/raqimbek/bankingutils/BankingUtilsApp.java

```diff
@@ -0,0 +1,26 @@
+package dev.andrylat.raqimbek.bankingutils;
+
+public class BankingUtilsApp {
+    private static CardValidationUserInteraction userInteraction = new CardValidationUserInteraction();
+    private static CardValidator cardValidator = new CardValidator();
+    private static PaymentSystemDeterminer paymentSystemDeterminer = new PaymentSystemDeterminer();
+
+    public static void main(String[] args) {
+        run();
+    }
+
+    public static void run() {
+    	userInteraction.write("Hello. Enter card number for validation:", System.out);
+    	userInteraction.read();        
+
+    	var cardValidationInfo = cardValidator.checkCardNumber(userInteraction.getInput());
+
+    	if (cardValidationInfo.getErrors().size() > 0) {
+    	    userInteraction.write(cardValidationInfo.errors.toString(), System.out);
+    	} else {
+    	    userInteraction.write(paymentSystemDeterminer
```

Comment:
You did the same mistake everywhere: readin one huge set of statements is very complex.

You need to do something like that:
```
var paymentSystem = paymentSystemDeterminer..determinePaymentSystemByCardNumber(userInteraction.getInput()).get().toString();
userInteraction.write(paymentSystem, System.out);
```

You can read it much more easier.